### PR TITLE
Add support for current python version.

### PR DESCRIPTION
**Pull request by sallner**
Changing testing infrastructure to tox for more convenience.

### CODE REVIEW

**Comment by reinout** [comment]
Looks fine to me!

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+commands =
+    zope-testrunner --test-path=. []
+deps =
+    .[test]
+
+[testenv:pypy3]
+commands =
+    zope-testrunner --test-path=. []
+deps =
+    .[test]
```
Comment (mgedmin) [inline tox.ini]: You do not need to repeat these settings: they're automatically inherited from the base `[testenv]` section.  All you need is overrides.

File: .coveragerc [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+precision = 2
+
+[html]
+directory = htmlcov
+
+[paths]
+source =
+   z3c/recipe/mkdir
+   .tox/*/lib/python*/site-packages/z3c/recipe/mkdir
+   .tox/pypy*/site-packages/z3c/recipe/mkdir
```
Comment (mgedmin) [inline .coveragerc:15]: I don't think this is necessary?  You only need this when you use `coverage combine` to combine the coverages of several runs (that tracked the source files in multiple directories).

OTOH it doesn't hurt either, and maybe somebody someday will want to use `coverage combine`.  *shrug*

**Review by mgedmin — APPROVED** [review]
LGTM

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    py27,py34,py35,py36,py37,pypy,pypy3,coverage
+
+[testenv]
+commands =
+    zope-testrunner --test-path=. []
+deps =
+    .[test]
+
+[testenv:pypy3]
+setenv = PYTHONWARNINGS = ignore::DeprecationWarning:distutils
```
Comment (mgedmin) [inline tox.ini:12]: (I am mildly curious about this: first it was simplejson, now distutils?  What is this warning that's only shown on pypy3?)

**Review by mgedmin — APPROVED** [review]

**Comment by reinout** [comment]
I've released it as 0.7.

File: tox.ini (hunk 3 above)
Comment (sallner) [reply tox.ini:12]: It looks something like this and spoils the doctest output on many occasions. But it does not seem to be present on travis. 

```
---------------------------------------------------------------------
File "/.../z3c.recipe.mkdir/.tox/pypy3/site-packages/z3c/recipe/mkdir/regression.txt", line 27, in regression.txt
Failed example:
    printx(system(join('bin', 'buildout')))
Expected:
    Installing mydir.
    mydir: created path: /sample-buildout/mypath1
    mydir:   mode 0750
    mydir: created path: /sample-buildout/mypath2
    mydir:   mode 0750
Got:
    Installing mydir.
    mydir: created path: /sample-buildout/mypath1
    mydir:   mode 0750
    mydir: created path: /sample-buildout/mypath2
    mydir:   mode 0750
    /home/USER/zope/z3c.recipe.mkdir/.tox/pypy3/lib-python/3/distutils/__init__.py:4: PendingDeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
      import imp
```

File: tox.ini (hunk 1 above)
Comment (sallner) [reply tox.ini]: I changed that.

File: .coveragerc (hunk 2 above)
Comment (sallner) [reply .coveragerc:15]: I kept it that way.

**Comment by sallner** [comment]
Thanks for the quick release.

**Pull request merged**